haystack/nodes/prompt/prompt_node.py

```diff
@@ -897,21 +897,25 @@ def run(
         :param meta: The meta to be used for the prompt. Usually not used.
         :param invocation_context: The invocation context to be used for the prompt.
         """
+        # The invocation context overwrites locals(), so if for example invocation_context contains a
+        # modified list of Documents under the `documents` key, such list is used.
+        invocation_context = {**locals(), **(invocation_context or {})}
+        invocation_context.pop("self")
```

Comment:
As a non-Python expert, I never liked these locals as slight refactoring and change of scope can cause issues here. But I am also okay if other reviewers are okay with the approach. I'd instead spell these variables out as we have a limited number of them, and we don't need to pop self.

Reply:
Fair enough! Fixing this :+1: